src/General-Rules-Tests/ReAsClassRuleTest.class.st

```diff
@@ -0,0 +1,32 @@
+Class {
+	#name : 'ReAsClassRuleTest',
+	#superclass : 'ReAbstractRuleTestCase',
+	#classInstVars : [
+		'critiques'
+	],
+	#category : 'General-Rules-Tests-Migrated',
+	#package : 'General-Rules-Tests',
+	#tag : 'Migrated'
+}
+
+{ #category : 'converting' }
+ReAsClassRuleTest >> myMethodCallingAsClass [
+	 ^ #ReAsClassRuleTest asClass
+]
+
+{ #category : 'converting' }
+ReAsClassRuleTest >> myMethodCallingAsClassIfAbsent [
+	 ^ #ReAsClassRuleTest asClassIfAbsent: [].
+]
+
+{ #category : 'converting' }
+ReAsClassRuleTest >> myMethodCallingAsClassIfPresents [
+	 ^ #ReAsClassRuleTest asClassIfPresent: [].
+]
+
+{ #category : 'tests' }
+ReAsClassRuleTest >> testRule [ 
+	|critiques|
+	critiques := self myCritiquesOnMethod: self myMethodCallingAsClassIfPresents.
+	self assert: critiques size equals: 1
```

Comment:
This one does not work as we have seen but we'll check together how we can fix that.